### PR TITLE
phase 1.1 codetrek module

### PR DESCRIPTION
**Pull request by nishant22029**
Targets #2736 

### Description
Created a new module in Portal Project with the name CodeTrek ,to track and capture journey of all the interns in Portal. This will help us to monitor the progress of each individual and see where we can or they can do better.

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: resources/views/layouts/navbar.blade.php [hunk 1]
```diff
@@ -148,6 +148,12 @@ class="caret"></span>
         </li>
     @endcan
 
+    @can('codetrek.view')
```
Comment (GauravGusain98) [inline resources/views/layouts/navbar.blade.php:151]: You have added this check for permission but we don't have any such permission yet, so this option will not available for users other than superadmin. For now please remove this check

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please remove package.lock.json as this is used only when you have added any new package.

File: resources/views/layouts/navbar.blade.php (hunk 1 above)
Comment (nishant22029) [reply resources/views/layouts/navbar.blade.php:151]: ok sir

**Comment by GauravGusain98** [comment]
The `package.lock.json` is not removed yet.

File: resources/views/layouts/navbar.blade.php (hunk 1 above)
Comment (nishant22029) [reply resources/views/layouts/navbar.blade.php]: sir should i have to remove package.json file from codetrek module

**Review by GauravGusain98 — APPROVED** [review]
LGTM

**Pull request merged**